app/electron/port-forward.ts

```diff
@@ -0,0 +1,52 @@
+import * as k8s from '@kubernetes/client-node';
+import { dialog, ipcMain } from 'electron';
+import { BrowserWindow } from 'electron/main';
+import * as net from 'net';
+
+export function handlePortForward(
+  mainWindow: BrowserWindow,
+  kubeconfig,
+  namespace: string,
+  podname: string,
+  targetPort: number
+) {
+  const kc = new k8s.KubeConfig();
+  if (kubeconfig) {
+    kc.loadFromFile(kubeconfig);
+  } else {
+    kc.loadFromDefault();
+  }
```

Comment:
I'm slightly concerned we're reading from the kubeconfig here, not only creating 2 points where we do this (the other is the server), but also we'd need to make sure that the cluster we're requesting the port-forward for is matched in the kubeconfig (which can be a problem with dynamic clusters).

I see that internally the lib uses an [API endpoint in the k8s server](https://github.com/kubernetes-client/javascript/blob/6c5a58f51b8cdb01c59cbd5b433013beb4940df2/src/portforward.ts#L44) and in the PortForward constructor we can give it a WebSocketHandler which we could create using a cooked up KubeConfig (dynamically, from the info we have).
Not sure if doing this is better than adding a special forward in the backend (because now it's already forwarding any calls we want, so it'd be a matter of using a different port + handler for that), but I think we'll find a good solution that doesn't involve re-reading the kubeconfig.

Reply:
updated this we now read from the backend api